File: scratch/extract.py
import sys
from bs4 import BeautifulSoup

def extract_data():
    with open(r"C:\Users\Mediacell\.gemini\antigravity-ide\brain\e6e90666-ee9f-4a65-ac7a-3ec600c66cfa\.system_generated\steps\1491\content.md", "r", encoding="utf-8") as f:
        html = f.read()

    soup = BeautifulSoup(html, 'html.parser')
    
    # Let's just find all the text and links to see the structure
    institutions = []
    for a in soup.find_all('a'):
        text = a.get_text(strip=True)
        href = a.get('href', '')
        if text and href and ('rv' in text.lower() or 'college' in text.lower() or 'institute' in text.lower() or 'school' in text.lower() or 'university' in text.lower()):
            if text not in institutions:
                institutions.append(text)
                print(f"Inst: {text} - Link: {href}")

if __name__ == "__main__":
    extract_data()

File: scratch/extract.js
const fs = require('fs');

const html = fs.readFileSync(String.raw`C:\Users\Mediacell\.gemini\antigravity-ide\brain\e6e90666-ee9f-4a65-ac7a-3ec600c66cfa\.system_generated\steps\1491\content.md`, 'utf8');

// Regex to find <a> tags and extract href and text
const regex = /<a\s+(?:[^>]*?\s+)?href=(["'])(.*?)\1[^>]*?>(.*?)<\/a>/gi;
let match;
const institutions = [];

while ((match = regex.exec(html)) !== null) {
    const href = match[2];
    const rawText = match[3].replace(/<[^>]+>/g, '').trim(); // Remove inner HTML tags
    if (rawText && (rawText.toLowerCase().includes('rv') || rawText.toLowerCase().includes('college') || rawText.toLowerCase().includes('institute') || rawText.toLowerCase().includes('school') || rawText.toLowerCase().includes('university'))) {
        if (!institutions.find(i => i.text === rawText)) {
            institutions.push({ text: rawText, link: href });
            console.log(`Inst: ${rawText} - Link: ${href}`);
        }
    }
}
